### PR TITLE
Amend check mode

### PR DESCRIPTION
**Pull request by Jennit07**
I think I have captured everything here, however, if I have missed any, feel free to correct me. This now includes `...` as a parameter and will read the default file path as `.rds`. We can specify `ext = "zsav"` if we want to save or read the file in SPSS format and also specify `check_mode = "write"` if we want to write the file to disk. These file path functions are set to `read` by default as this is passed through the `get_file_path` function.

Scripts should now be in this format for writing to disk: 

SPSS: 
```r
# Reading
haven::read_sav(get_source_extract_path(year, "Outpatient", ext = "zsav"))
# Writing
haven::write_sav(get_source_extract_path(year, "Outpatient", ext = "zsav", check_mode = "write"), compress = "zsav")
```

R: 
```r
# Reading
readr::read_rds(get_source_extract_path(year, "Outpatients"))
# Writing
readr::write_rds(get_source_extract_path(year, "Outpatients", check_mode = "write"), compress = "xz")
```

I have also amended the LTCs script on this branch as I was changing file paths and came across an error. This now includes a new function `get_ltcs_path()` in the same format as others here. There was a part of the code where this was breaking as an older object was specified while trying to do a save. This has been amended and should now work.

### CODE REVIEW

File: Production_scripts/Create_slf_lookups/lookup_ltc.R [hunk 1]
```diff
@@ -49,55 +48,50 @@ ltc_file <- read_csv(
...
   )
 
-## Create LTC flags 1/0 ##
+# Create LTC flags 1/0------------------------------------
 
 # Set flags to 1 or 0 based on FY
 # then sort by chi
 
 ltc_flags <- ltc_file %>%
   mutate(across(ends_with("date"), list(flag = ~ if_else(is.na(.x) | .x > end_fy, 0, 1)))) %>%
```
Comment (Moohan) [inline Production_scripts/Create_slf_lookups/lookup_ltc.R:83]:
```suggestion
  mutate(across(ends_with("date"), list(flag = ~ if_else(is.na(.x) | .x > end_fy, 0L, 1L)))) %>%
```

**Review by catherineholland1 — APPROVED** [review]
all looks good

File: Production_scripts/Create_slf_lookups/lookup_gp_practice.R [hunk 2]
```diff
@@ -84,12 +70,20 @@ pc <- read_spd_file() %>%
...
+  opendata %>%
+  ## match cluster information onto the practice reference list ##
+  left_join(gpprac_file, by = c("gpprac", "postcode")) %>%
+  # sort by postcode
+  arrange(postcode) %>%
+  # match on geography info - postcode
+  left_join(pc_lookup, by = "postcode") %>%
   # rename hb2018
   rename(hbpraccode = "hb2018") %>%
+  # order variables
```
Comment (Moohan) [inline Production_scripts/Create_slf_lookups/lookup_gp_practice.R:86]: I think `postcode` here refers to the postcode of the GP Practice? If so could we rename it to `gpprac_postcode` for clarity/readability, please?

File: Production_scripts/Create_social_care_lookups/lookup_sc_demographics.R [hunk 3]
```diff
@@ -141,9 +141,9 @@ outfile <-
 
 ## save file ##
 # .zsav file
-haven::write_sav(outfile, get_sc_demog_lookup_path(), compress = TRUE)
+haven::write_sav(outfile, get_sc_demog_lookup_path(check_mode = "write"), compress = TRUE)
```
Comment (Moohan) [inline Production_scripts/Create_social_care_lookups/lookup_sc_demographics.R:144]:
```suggestion
haven::write_sav(outfile, get_sc_demog_lookup_path(ext = "zsav", check_mode = "write"), compress = TRUE)
```

File: R/get_source_extract_path.R [hunk 4]
```diff
@@ -25,7 +25,7 @@ get_source_extract_path <- function(year,
                                       "Outpatients",
                                       "PIS"
                                     ),
-                                    ext = c("zsav", "rds")) {
+                                    ...) {
   type <- match.arg(type)
   ext <- match.arg(ext)
```
Comment (Moohan) [inline R/get_source_extract_path.R:30]:
```suggestion
```

**Review by Moohan — CHANGES_REQUESTED** [review]
It looks like the Dev SOP is being deleted for some reason, could you do:
```
git fetch --all
git checkout origin/main-R -- R_development_SOP.md`
```
which should make sure you have the latest version on your branch.

Other than my comments it looks good! I think if we've missed/broken anything else we'll pick it up at some point soon on some other test and review process.

**Comment by Jennit07** [comment]
To do: I need to go back to this one and make the changes suggested

**Comment by Moohan** [comment]
I re-requested a review from @catherineholland1, as I just dealt with all the merge conflicts and made a couple of other changes based on my previous comments.

**Review by catherineholland1 — APPROVED** [review]
ready for merge :)

**Pull request merged**